src/test/ensemble_config_change_stress.cpp

```diff
@@ -0,0 +1,668 @@
+                SPDLOG_TRACE("Create:[{}]={}:{}", static_cast<uint>(retCode), ovms::Status(retCode).string(), counter);
+            }
+        }
+        for (auto& [retCode, counter] : executePipelineRetCodesCounters) {
+            if (counter > 0) {
+                SPDLOG_TRACE("Execute:[{}]={}:{}", static_cast<uint>(retCode), ovms::Status(retCode).string(), counter);
+            }
+        }
+        SPDLOG_INFO("Executed: {} inferences", stressIterationsLimit - stressIterationsCounter);
+        EXPECT_GT(stressIterationsCounter, 0) << "Reaching 0 means that we might not test enough \"after config change\" operation was applied";
+    }
+};
+
+TEST_F(StressPipelineConfigChanges, AddNewVersionDuringPredictLoad) {
+    bool performWholeConfigReload = false;                            // we just need to have all model versions rechecked
+    std::set<StatusCode> requiredCreatePDResults = {StatusCode::OK};  // we expect full continuouity of operation
+    std::set<StatusCode> requiredExecuteResults = {StatusCode::OK};   // we expect full continuouity of operation
+    std::set<StatusCode> allowedCreatePDResults = {};
+    std::set<StatusCode> allowedExecuteResults = {};
+    performStressTest(&StressPipelineConfigChanges::defaultVersionRemove,
+        performWholeConfigReload,
+        requiredCreatePDResults,
+        requiredExecuteResults,
+        allowedCreatePDResults,
+        allowedExecuteResults);
+}
+TEST_F(StressPipelineConfigChanges, RemoveDefaultVersionDuringPredictLoad) {
+    std::set<StatusCode> requiredCreatePDResults = {StatusCode::OK,
+        StatusCode::PIPELINE_DEFINITION_NOT_LOADED_YET};            // we hit when all config changes finish to propagate
+    std::set<StatusCode> requiredExecuteResults = {StatusCode::OK,  // the model did not unload yet
+        StatusCode::MODEL_VERSION_NOT_LOADED_ANYMORE,               // we hit default version which is unloaded already but default is not changed yet
+        StatusCode::MODEL_VERSION_MISSING};                         // there is no default version since all are either not loaded properly or retired
+    // if we do not hit different execute results than OK it means that the stress was not really stress
+    std::set<StatusCode> allowedCreatePDResults = {};
+    std::set<StatusCode> allowedExecuteResults = {};
+    // we need whole config reload since there is no other way to dispose
+    // all model versions different than removing model from config
+    bool performWholeConfigReload = true;
+    performStressTest(&StressPipelineConfigChanges::defaultVersionRemove,
+        performWholeConfigReload,
+        requiredCreatePDResults,
+        requiredExecuteResults,
+        allowedCreatePDResults,
+        allowedExecuteResults);
+}
+TEST_F(StressPipelineConfigChanges, ChangeToShapeAutoDuringPredictionLoad) {
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredCreatePDResults = {StatusCode::OK};  // we expect full continuouity of operation
```

Comment:
Shouldnt we receive errors when shape is changed to auto?

Reply:
It is not allowed transition so we should not expect errors while doing such change.